tools/undo-block/src/cli.rs

```diff
@@ -0,0 +1,75 @@
+use chrono::Utc;
+use near_chain::types::LatestKnown;
+use near_chain::{ChainStore, ChainStoreAccess, ChainStoreUpdate};
+use near_chain_configs::GenesisValidationMode;
+use near_primitives::block::Tip;
+use near_primitives::utils::to_timestamp;
+use near_store::{Mode, NodeStorage};
+use nearcore::load_config;
+use std::path::Path;
+
+#[derive(clap::Parser)]
+pub struct UndoBlockCommand {}
+
+impl UndoBlockCommand {
+    pub fn run(
+        self,
+        home_dir: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        let near_config = load_config(home_dir, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+
+        let cold_config: Option<&near_store::StoreConfig> = near_config.config.cold_store.as_ref();
+        let store_opener = NodeStorage::opener(
+            home_dir,
+            near_config.config.archive,
+            &near_config.config.store,
+            cold_config,
+        );
+
+        let storage = store_opener.open_in_mode(Mode::ReadWrite).unwrap();
```

Comment:
when I compile nearcore with `--features=protocol_feature_flat_state`, and when I run the undo-block tool, I get this error for this line: 
`thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: IO(Custom { kind: Other, error: "Invalid argument: Column families not opened: FlatStorageStatus, FlatStateDeltaMetadata, FlatStateChanges, FlatState" })', tools/undo-block/src/cli.rs:31:66
`
How to address this?